signer/tests/integration/transaction_coordinator.rs

```diff
@@ -0,0 +1,331 @@
+                    Ok(AccountInfo {
+                        balance: 0,
+                        locked: 0,
+                        unlock_height: 0,
+                        // The nonce is used to create the stacks tx
+                        nonce,
+                    })
+                })
+            });
+
+            // Dummy value
+            client
+                .expect_estimate_fees()
+                .once()
+                .returning(move |_, _| Box::pin(async move { Ok(25505) }));
+        })
+        .await;
+
+    let num_signers = 7;
+    let signing_threshold = 5;
+    let context_window = 10;
+
+    let network = network::in_memory::Network::new();
+    let signer_info = testing::wsts::generate_signer_info(&mut rng, num_signers);
+
+    let mut testing_signer_set =
+        testing::wsts::SignerSet::new(&signer_info, signing_threshold, || network.connect());
+
+    let (aggregate_key, bitcoin_chain_tip) =
+        run_dkg(&context, &mut rng, &mut testing_signer_set).await;
+
+    // Ensure we have a signers UTXO (as a donation, to not mess with the current
+    // temporary `get_swept_deposit_requests` implementation)
+    push_utxo_donation(&context, &aggregate_key, &setup.sweep_block_hash).await;
+
+    assert_eq!(
+        context
+            .get_storage()
+            .get_swept_deposit_requests(&bitcoin_chain_tip.block_hash, context_window)
+            .await
+            .expect("failed to get swept deposits")
+            .len(),
+        1
+    );
+
+    let (broadcasted_transaction_tx, _broadcasted_transaction_rxeiver) =
+        tokio::sync::broadcast::channel(1);
+
+    // This task logs all transactions broadcasted by the coordinator.
+    let mut wait_for_transaction_rx = broadcasted_transaction_tx.subscribe();
+    let wait_for_transaction_task =
+        tokio::spawn(async move { wait_for_transaction_rx.recv().await });
+
+    // Setup the stacks client mock to broadcast the transaction to our channel.
+    context
+        .with_stacks_client(|client| {
+            client.expect_submit_tx().times(1..).returning(move |tx| {
+                let tx = tx.clone();
+                let txid = tx.txid();
+                let broadcasted_transaction_tx = broadcasted_transaction_tx.clone();
+                Box::pin(async move {
+                    broadcasted_transaction_tx
+                        .send(tx)
+                        .expect("Failed to send result");
+                    Ok(SubmitTxResponse::Acceptance(txid))
+                })
+            });
+        })
+        .await;
+
+    // Get the private key of the coordinator of the signer set.
+    let private_key = select_coordinator(&setup.sweep_block_hash.into(), &signer_info);
+
+    // TODO: should the tx coordinator evloop use its private key instead of `config().signer.private_key`?
+    let mut coordinator_ctx = context.clone();
+    coordinator_ctx.inner.config_mut().signer.private_key = private_key;
```

Comment:
Code ref: https://github.com/stacks-network/sbtc/pull/650/files#diff-35276625f928b2224c6df10468d8afee7dcf4168f01de937a30650e521b90bc2R439

With the updated impl (where the coordinator does not sign the tx) this is relevant only if num_signers is 1. Still, as we pass the `private_key` to `TxCoordinatorEventLoop`, should we use that one instead of loading it from the config?

Reply:
They should be the same. We should remove the field from both `TxCoordinatorEventLoop` and `TxSignerEventLoop`, and always use the one in the config (just not in this PR). 

I would remove this line and the corresponding `config_mut` method that was added to the context.